bitacross-worker/bitacross/core/bc-task-processor/src/lib.rs

```diff
@@ -0,0 +1,651 @@
+	H::StateT: SgxExternalitiesTrait,
+	Responder: SendRpcResponse<Hash = H256>,
+{
+	#[allow(clippy::too_many_arguments)]
+	pub fn new(
+		shielding_key: Arc<SKR>,
+		signing_key_access: Arc<SIGNINGAK>,
+		ethereum_key_repository: Arc<EKR>,
+		bitcoin_key_repository: Arc<BKR>,
+		enclave_signer: Arc<S>,
+		state_handler: Arc<H>,
+		ocall_api: Arc<O>,
+		relayer_registry_lookup: Arc<RRL>,
+		enclave_registry_lookup: Arc<ERL>,
+		signer_registry_lookup: Arc<SRL>,
+		signing_key_pub: [u8; 32],
+		ceremony_registry: Arc<RwLock<CeremonyRegistry<BKR>>>,
+		ceremony_command_tmp: Arc<RwLock<CeremonyCommandTmp>>,
+		responder: Arc<Responder>,
+	) -> Self {
+		Self {
+			shielding_key,
+			signing_key_access,
+			ethereum_key_repository,
+			bitcoin_key_repository,
+			enclave_signer,
+			state_handler,
+			ocall_api,
+			relayer_registry_lookup,
+			enclave_registry_lookup,
+			signer_registry_lookup,
+			signing_key_pub,
+			ceremony_registry,
+			ceremony_command_tmp,
+			responder,
+		}
+	}
+}
+
+#[allow(clippy::type_complexity)]
+pub fn run_bit_across_handler_runner<SKR, SIGNINGAK, EKR, BKR, S, H, O, RRL, ERL, SRL, Responder>(
+	context: Arc<BitAcrossTaskContext<SKR, SIGNINGAK, EKR, BKR, S, H, O, RRL, ERL, SRL, Responder>>,
+) where
+	SKR: AccessKey + AccessPubkey<KeyType = Rsa3072PubKey> + Send + Sync + 'static,
+	SIGNINGAK: AccessKey<KeyType = ed25519::Pair> + Send + Sync + 'static,
+	EKR: AccessKey<KeyType = EcdsaPair> + Send + Sync + 'static,
+	BKR: AccessKey<KeyType = SchnorrPair> + Send + Sync + 'static,
+	<SKR as AccessKey>::KeyType: ShieldingCryptoEncrypt + ShieldingCryptoDecrypt + 'static,
+	S: StfEnclaveSigning<TrustedCallSigned> + Send + Sync + 'static,
+	H: HandleState + Send + Sync + 'static,
+	H::StateT: SgxExternalitiesTrait,
+	O: EnclaveOnChainOCallApi + EnclaveMetricsOCallApi + EnclaveAttestationOCallApi + 'static,
+	RRL: RelayerRegistryLookup + Send + Sync + 'static,
+	ERL: EnclaveRegistryLookup + Send + Sync + 'static,
+	SRL: SignerRegistryLookup + Send + Sync + 'static,
+	Responder: SendRpcResponse<Hash = H256> + Send + Sync + 'static,
+{
+	// timeout tick
+	let ceremony_registry = context.ceremony_registry.clone();
+	let ceremony_command_tmp = context.ceremony_command_tmp.clone();
+	let time_to_live = 30u64;
+	std::thread::spawn(move || loop {
+		std::thread::sleep(Duration::from_secs(3));
+		let now = get_current_timestamp();
+		{
+			let mut ceremony_registry_write = ceremony_registry.write().unwrap();
+			ceremony_registry_write
+				.retain(|_, &mut (_, create_time)| now - create_time < time_to_live);
+		}
+		{
+			let mut command_tmp_write = ceremony_command_tmp.write().unwrap();
+			command_tmp_write.retain(|_, &mut (_, create_time)| now - create_time < time_to_live);
+		}
+	});
+
+	let bit_across_task_receiver = init_bit_across_task_sender_storage();
+	let peers_map = Arc::new(Mutex::new(HashMap::<[u8; 32], DirectRpcClient>::new()));
+	let command_threads_count = 4;
```

Comment:
Can this be parametrized through CLI args ?
It would be good to have an opportunity to tweak it without preparing new binary.

Reply:
It's better to do so, but could we improve this in another PR? I'm not familiar with code about passing configuration parameters for now.

Reply:
I'm fine with it as long as thread usage won't cause TCS crash on prod.